gtsam/nonlinear/NonlinearFactorGraph.cpp

```diff
@@ -61,19 +61,26 @@ void NonlinearFactorGraph::print(const std::string& str, const KeyFormatter& key
 
 /* ************************************************************************* */
 void NonlinearFactorGraph::printErrors(const Values& values, const std::string& str,
-                                       const KeyFormatter& keyFormatter) const {
+    const KeyFormatter& keyFormatter,
+    const std::function<bool(const Factor* /*factor*/, double /*whitenedError*/, size_t /*index*/)>& printCondition) const
```

Comment:
This should be optional

Reply:
Actually, that was my first implementation, using `boost::optional`. But it had problems to automatically bind to a lambda as input.
Will give it another try...

Thanks!

Reply:
Actually I think simple overloading here makes perfect sense :)

Reply:
BTW, this https://godbolt.org/z/aSio9A also work